File: flask-application/issues/path_traversal.py
from flask import Blueprint, request

path_traversal = Blueprint('path_traversal', __name__)


@path_traversal.route("/path-traversal")
def get_example_data():
    category = request.args.get("category")
    try:
        with open(f"../assets/public/{category}", "r") as file:
            data = file.read()
            return data
    except FileNotFoundError:
        return "Unknown category, please try 'artists' or 'genres'", 404
    except Exception:
        return "An unexpected error occurred, please try again later", 500

File: laravel-application/routes/issues/path-traversal.php
<?php

use Illuminate\Support\Facades\Route;

Route::get('/', function () {
  $category = request()->query('category');
  $filePath = realpath(base_path('../assets/public/') . $category);

  try {
    $data = file_get_contents($filePath);
    return response($data);
  } catch (\Exception $exception) {
    if (strpos($exception->getMessage(), 'file_get_contents(): Read of') !== false) {
      return response("Unknown category, please try 'artists' or 'genres'", 404);
    }
    return response("An unexpected error occurred, please try again later", 500);
  }
});
